docs/introduction/1-gettingstarted.rst

```diff
@@ -95,6 +95,113 @@ To serve static files from a directory, set the ``branch`` keyword argument on t
+whatever slots you want in your page template.
+
+You can also specify a dictionary of default values to fill slots with.
+
+Next, you want to create a route that is plated with that ``Plating``, by using
+the ``Plating.routed`` decorator.  ``@style.routed`` takes a route from the
+Klein instance, in this case ``app``, and then a template for the content
+portion (the ``Plating.CONTENT`` slot) of the page.  The decorated function
+must then return a dictionary of the values to populate the slots in the
+template with.
+
+Let's start with a really simple page that just has a static template to fill
+the content slot.
+
+.. literalinclude:: codeexamples/html.py
+    :lines: 26-37
+
+This page generates some links to various sub-pages which we'll get to in a
+moment.  But first, if you load ``http://localhost:8080/``, you'll see that the
+template specified for ``root`` is inserted at the point in the template for
+``style`` specified the content should go.
+
+Next, we should actually try injecting some data.
+
+.. literalinclude:: codeexamples/html.py
+    :lines: 39-50
+
+Here you can see the ``/foods/...`` route for showing information about a food.
+In the content template, we've got slots for ``"name"``, ``"rating"``, and
+``"carbohydrates"``, the three primary properties which define a food.  The
+decorated function then returns a dictionary that returns values for each of
+those slots, as well as a value for "title".
+
+Each of these slots is only filled with a single item, though.  What if you
+need to put multiple items into the template?  The route for ``/places/...``
+can show us:
+
+.. literalinclude:: codeexamples/html.py
+    :lines: 52-68
+
+Here you can see the special ``<slotname>:list`` renderer in use.  By
+specifying the ``render=`` attribute of a tag (in this case, a ``li`` tag) to
+be ``foods:list``, we invoke a ``twisted.web.template`` renderer that repeats
+the tag it is the renderer for, inserting each element of that list into the
+special ``"item"`` slot.
+
+You can view each of these pages in a web browser now, and you can see their
+contents; we've built a little website that generates random values for these
+types of data.  But we've *also* built a JSON API.  If you access, for example,
+``http://localhost:8080/places/chicago``, you'll see an HTML view, but if you
+add the query parameter ``json=1``
+(e.g. ``http://localhost:8080/places/chicago?json=1``) you will see a JSON
+result like this:
+
+.. code-block:: json
+
+    {
+        "foods": [
+            "pizza",
+            "cheeseburgers",
+            "hot dogs"
+        ],
+        "latitude": -32.610538480748815,
+        "longitude": -9.38433633489143,
+        "name": "chicago",
+        "title": "Place: chicago"
+    }
+
+Any route decorated by ``@routed`` will similarly give you structured data if
+you ask for it via ``?json=1``, so you can build your JSON API and your HTML
+frontend at the same time.
```

Comment:
Excellent narrative documentation!!  it's missing discussions  of/examples for `presentation_slots` and widgets, though.